bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeConverter.java

```diff
@@ -923,6 +923,10 @@ public static int stringToByte(String value) throws NumberFormatException, BErro
     }
 
     public static Double stringToFloat(String value) throws NumberFormatException {
+        String upperCaseValue = value.toUpperCase();
+        if (upperCaseValue.endsWith("F") || upperCaseValue.endsWith("D")) {
```

Comment:
Shall we change this check such that it will get the last character of a non empty string and check whether it is a `f` or `F` or `d` or `D` using switch case

Reply:
Need to change the other occurrences same as this.